generator/final_processor_intermediate_mwm.cpp

```diff
@@ -0,0 +1,519 @@
+#include "generator/final_processor_intermediate_mwm.hpp"
+
+#include "generator/affiliation.hpp"
+#include "generator/booking_dataset.hpp"
+#include "generator/city_boundary_processor.hpp"
+#include "generator/feature_builder.hpp"
+#include "generator/feature_merger.hpp"
+#include "generator/promo_catalog_cities.hpp"
+#include "generator/type_helper.hpp"
+
+#include "indexer/classificator.hpp"
+
+#include "platform/platform.hpp"
+
+#include "base/assert.hpp"
+#include "base/file_name_utils.hpp"
+#include "base/geo_object_id.hpp"
+#include "base/string_utils.hpp"
+#include "base/thread_pool_computational.hpp"
+#include "base/thread_pool_delayed.hpp"
+
+#include <algorithm>
+#include <cstdint>
+#include <cstdio>
+#include <future>
+#include <iterator>
+#include <memory>
+#include <tuple>
+#include <unordered_set>
+#include <vector>
+
+#include "defines.hpp"
+
+using namespace base::thread_pool;
+using namespace feature;
+using namespace serialization_policy;
+
+namespace generator
+{
+namespace
+{
+template <typename ToDo>
+void ForEachCountry(std::string const & temporaryMwmPath, ToDo && toDo)
+{
+  Platform::FilesList fileList;
+  Platform::GetFilesByExt(temporaryMwmPath, DATA_FILE_EXTENSION_TMP, fileList);
+  for (auto const & filename : fileList)
+    toDo(filename);
+}
+
+std::vector<std::vector<std::string>> GetAffilations(std::vector<FeatureBuilder> const & fbs,
+                                                     AffiliationInterface const & affiliation,
+                                                     size_t threadsCount)
+{
+  computational::ThreadPool pool(threadsCount);
+  std::vector<std::future<std::vector<std::string>>> futuresAffilations;
+  for (auto const & fb : fbs)
+  {
+    auto result = pool.Submit([&]() {
+      return affiliation.GetAffiliations(fb);
+    });
+    futuresAffilations.emplace_back(std::move(result));
+  }
+
+  std::vector<std::vector<std::string>> resultAffilations;
+  resultAffilations.reserve(futuresAffilations.size());
+  for (auto & f : futuresAffilations)
+    resultAffilations.emplace_back(f.get());
+
+  return resultAffilations;
+}
+
+// Records |fbs| according to |affiliations|.
```

Comment:
Writes |fbs| to countries tmp.mwm files |fbs| belongs to according to |affiliations|.

Reply:
++